lib/foreman_maintain/concerns/base_database.rb

```diff
@@ -57,20 +58,25 @@ def psql(query, config = configuration)
         if ping(config)
           execute(psql_command(config),
             :stdin => query,
-            :hidden_patterns => [config['password']])
+            :hidden_patterns => [config['password']],
+            :config => user)
         else
           raise_service_error
         end
       end
 
       def ping(config = configuration)
-        execute?(psql_command(config),
-          :stdin => 'SELECT 1 as ping',
-          :hidden_patterns => [config['password']])
+        execute?(psql_command(config) + ' -c "SELECT 1 as ping"',
+          :hidden_patterns => [config['password']],
+          :user => config['user'])
       end
 
       def dump_db(file, config = configuration)
-        execute!(dump_command(config) + " > #{file}", :hidden_patterns => [config['password']])
+        if config['user']
+          execute!("chown -R #{config['user']}:#{config['user']} #{File.dirname(file)}")
+        end
+        execute!(dump_command(config) + " -f #{file}",
+          :hidden_patterns => [config['password']], :user => config['user'])
       end
```

Comment:
Right now, this relies on the DB credentials we parse from the config and in the super special case of gateway, the shell user of the proxy.

For restores, we have this "if local, use `postgres` user, else the parsed config as `root`", if we use the same logic for dumping, we reduce the special-casing for gateway (as we either use `postgres`, which has access to all databases anyway, or a remote connection that doesn't require `foreman-proxy` user).

Reply:
When I wrote this I didn't think I could use the postgres user for some reason. Using `postgres` would make the code here a whole lot simpler :+1: